chore: clean up contract verify flow

ludamad opened the pull request:
Contract verification is now enabled only when deploying to mainnet or sepolia AND `ETHERSCAN_API_KEY` env var is set
We error out early if VERIFY_CONTRACTS is set and we would not be able to deploy. but essentially the only purpose VERIFY_CONTRACTS has now is to enable this as a hard error.

Also, we refactor logic in forge deploys to reduce duplication

just-mitch commented on spartan/scripts/deploy_network.sh:
```diff
@@ -272,9 +272,12 @@ fi
 if [[ "${VERIFY_CONTRACTS:-}" == "true" && "${CREATE_ROLLUP_CONTRACTS}" == "true" ]]; then
-  if [ -z "$ETHERSCAN_API_KEY" ]; then
-    echo "Error: ETHERSCAN_API_KEY is not set, but VERIFY_CONTRACTS=true. Cannot verify contracts without Etherscan API key (i.e. we need API access to the verification service)."
+  if [[ -z "${ETHERSCAN_API_KEY:-}" ]]; then
```
turbo nit: could be one `if` right?

ludamad committed:
chore: clean up contract verify flow

Contract verification is now enabled only when deploying to mainnet or sepolia AND `ETHERSCAN_API_KEY` env var is set
We error out early if VERIFY_CONTRACTS is set and we would not be able to deploy. but essentially the only purpose VERIFY_CONTRACTS has now is to enable this as a hard error.

Also, we refactor logic in forge deploys to reduce duplication
